тз 16 Докер

RomanAntipenko opened the pull request:
feat: Разбей приложение на несколько модулей Gateway, server
feat: Распиши для каждого из них свой Dockerfile, pom, application.properties
feat: Теперь шлюз производит некоторую валидацию входных данных, а сервер занят бизнес логикой
feat: согласно docker-compose.yaml  и общему pom приложенния собирается из трех коонтейнеров на соответствующих портах
fix: произведи соответствующий рефакторинг
fix: подправь тесты

RomanAntipenko committed:
feat: Разбей приложение на несколько модулей Gateway, server
feat: Распиши для каждого из них свой Dockerfile, pom, application.properties
feat: Теперь шлюз производит некоторую валидацию входных данных, а сервер занят бизнес логикой
feat: согласно docker-compose.yaml  и общему pom приложенния собирается из трех коонтейнеров на соответствующих портах
fix: произведи соответствующий рефакторинг
fix: подправь тесты

RomanAntipenko committed:
feat: Добавь тесты для шлюза

RomanAntipenko committed:
del: Убрал лишнее

RomanAntipenko committed:
del: Убрал лишнее
style: чек стайл

stfaustoff commented on gateway/src/main/java/ru/practicum/shareit/booking/controller/BookingController.java:
```diff
@@ -0,0 +1,82 @@
+    private final BookingClient bookingClient;
+
+    @GetMapping
+    public ResponseEntity<Object> getBookingsForBookers(@RequestHeader("X-Sharer-User-Id") long userId,
```
_Предложение_
По своему опыту скажу что когда одна строка встречается очень часто, то я завожу специальный класс для констант и переношу строку туда. Это и документация и удобство в исправлении.

stfaustoff commented on gateway/src/main/java/ru/practicum/shareit/client/BaseClient.java:
```diff
@@ -0,0 +1,117 @@
+        this.rest = rest;
+    }
+
+    private static ResponseEntity<Object> prepareGatewayResponse(ResponseEntity<Object> response) {
```
Я знаю что этот метод уже был в класс, но на твой взгляд, есть ли преимущества в том что он статический?
И к тому же все приватные методы идут самыми последними

stfaustoff commented on gateway/pom.xml:
```diff
@@ -0,0 +1,70 @@
+    <name>ShareIt Gateway</name>
+
+    <dependencies>
+        <dependency>
```
Эта зависимость, а так же lombok, actuator и некоторые другие уже есть в родительском поме, так что здесь это прописывать не обязательно

stfaustoff commented on pom.xml:
```diff
@@ -57,15 +63,14 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-validation</artifactId>
```
Validation можно перенести в gateway, так как в другом модуле он не нужен.

stfaustoff commented on server/pom.xml:
```diff
@@ -0,0 +1,92 @@
+        </plugins>
+    </build>
+
+    <!--<profiles>
```
Закомментированные участки кода перед PR   лучше убирать

stfaustoff commented on server/pom.xml:
```diff
@@ -0,0 +1,92 @@
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+
+        <dependency>
```
Тут тоже повторяющиеся зависимости лучше поубирать.

RomanAntipenko committed:
fix: перераспределил зависимости в POM файла проекта
del: убрал все что было связанно с валидацией из модуля Server
fix: убрал закоментировный код
feat: добавил классы для констант
fix: убрал static у метода в классе BaseClient

RomanAntipenko committed:
fix: забыл про относительное время
